### PR TITLE
CVE-2010-4490 and CVE-2014-3198

### PR DESCRIPTION
**Pull request by agarrity**
Completed CVE-2010-4490 and CVE-2014-3198

### CODE REVIEW

File: cves/CVE-2010-4490.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description: 
+  A malformed video could cause the application to crash by creating an
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: Can you give me more details about the malformed video? Any particular part of the data?

Also, memory corruption is not JUST a DOS attack - discuss if there are other possibilities other than a crash.

File: cves/CVE-2010-4490.yml [hunk 2]
```diff
@@ -16,6 +16,10 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description:
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: Need the CWE too

File: cves/CVE-2010-4490.yml [hunk 3]
```diff
@@ -31,8 +35,14 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 4df400db387b816286d2a6b1519b9d4720e83393
-  :note: ''
-vccs: []
+  :note: 
+    The fix checked to make sure that the stream index was less than
+    the size of the array holding the steams.
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: spelling "steams"

File: cves/CVE-2010-4490.yml [hunk 4]
```diff
@@ -31,8 +35,14 @@ fixes_vcc_instructions: |
...
+vccs:
+- commit: 61374e50e3714c93ef130aea5f0b35cd4b8bca3e
+  note: 
+    This VCC was decently large and had other checks and unittesting, but none
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: "unit testing"

File: cves/CVE-2010-4490.yml [hunk 5]
```diff
@@ -72,9 +84,10 @@ discovered:
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
   answer: 
-  date: 
-  automated: 
-  google: 
+    The vulnerability was found by miaubiz@gmail.com who i beleive is an employee.
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: Why do you believe he's an employee? Anything else about how it was found? Who is this person? I need a bit more here.

File: cves/CVE-2010-4490.yml [hunk 6]
```diff
@@ -83,8 +96,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: Media filter in the media/filters directory
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: Look at the tags on the bug in the bug report too - "filters" is too narrow.

File: cves/CVE-2010-4490.yml [hunk 7]
```diff
@@ -83,8 +96,8 @@ subsystem:
...
+  name: filters
 interesting_commits:
   question: |
     Are there any interesting commits between your VCC(s) and fix(es)?
```
Comment (andymeneely) [inline cves/CVE-2010-4490.yml]: I'd like to see a discussion of interesting commits to show that you actually looked at these. If you don't find any, I would like an explanation as to why.

File: cves/CVE-2014-3198.yml [hunk 8]
```diff
@@ -16,6 +16,12 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description: 
+  An attempt to index by -1 which causes a buffer overflow.  A method that returns
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: In what? What part of Chrome is this in?

File: cves/CVE-2014-3198.yml [hunk 9]
```diff
@@ -16,6 +16,12 @@ description_instructions: |
...
+  the index of the first visible page will return -1 when there is no visible page.
+  This was not being checked for and could cause a potential denial of service 
+  when trying to index by -1 and causing a buffer overflow.
+  
+  This was fixed by checking if the index was -1 and returning if it was.
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: Buffer overflows are more than just denial of service. You're trusting the CVE description too much. Go over this one again.

File: cves/CVE-2014-3198.yml [hunk 10]
```diff
@@ -16,6 +16,12 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: I need the CWE

File: cves/CVE-2014-3198.yml [hunk 11]
```diff
@@ -55,8 +65,11 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer: 
-  code: 
-  fix: 
+    The fix did not invlove automated tests.  The bug was found with a fuzzer
+    and the fix was tested against the fuzzer.  Because there is test coverage
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: I wouldn't count the fuzzer as a unit test. Yes, it's an automated tester but not in the same way that I meant for the question. Look around the rest of the directory, this subsystem, and other commits to see if you can find the relevant unit tests. Look at the code review and for their build information. If you still can't find tests THEN, then write about that. Basically demonstrate to me that you looked into this one.

File: cves/CVE-2014-3198.yml [hunk 12]
```diff
@@ -73,9 +86,12 @@ discovered:
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
   answer: 
-  date: 
-  automated: 
-  google: 
+    The vulnerability was found by ClusterFuzz.  They could not reproduce
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: Can you report what kind of error ClusterFuzz reported? Segfault? Uninitialized read? Something more specific about the memory corruption.

File: cves/CVE-2014-3198.yml [hunk 13]
```diff
@@ -140,8 +156,8 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: they did not consider the cases where an invalid index may be returned.
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: Need some more context here. Tie it back to PDFs and what that index corresponds to.

File: cves/CVE-2014-3198.yml [hunk 14]
```diff
@@ -173,3 +189,6 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer: 
+    The vulnerability was caused by simply overlooking the case where there
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: I need more here. It's a simple mistake, I agree, but tie in more material from the course. Lack of tests? Integration between subsystems? Lots of churn? Releasing near a deadline? Simple code things have a bigger story usually.

File: cves/CVE-2014-3198.yml [hunk 15]
```diff
@@ -84,8 +100,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: Based on the description in the CVE
+  name: PDFium
 interesting_commits:
   question: |
```
Comment (andymeneely) [inline cves/CVE-2014-3198.yml]: Same thing as above for interesting commits. Show me that you did the git log thing.

File: cves/CVE-2010-4490.yml [hunk 16]
```diff
@@ -54,8 +64,10 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer: 
-  code: 
-  fix: 
+    There was unittesting for this code but it did not cover this vulnerability.
```
Comment (Justin-Nietzel) [inline cves/CVE-2010-4490.yml]: more "unit testing"

File: cves/CVE-2014-3198.yml (hunk 8 above)
Comment (Justin-Nietzel) [inline cves/CVE-2014-3198.yml]: What was the end result for this bug? Did it crash a tab, crash the application, make chrome freeze up for a bit? Would the user know that something bad occurred?

**Pull request merged**